Pima indians diabetes and linear classifier/Gradient Descent/task.md

```diff
@@ -0,0 +1,53 @@
+
+The `fit` method should return a list of values of the performance functional $Q(w)$ at each iteration of the gradient descent.
+
+As a stopping criterion, use the cutoff `threshold` at the distance between the weight vectors in the current and the preceding iteration.
+You can choose any distance, for example, Euclidean or `l1`.
+
+In the `precision_recall` file, you will find the functions for calculating precision and recall. You can use them to assess the quality of the classification.
+
+<div class="hint">
+Precision defines the reliability of the classifier. In our case, it will represent the proportion of anamneses for which type 2 diabetes was predicted <b>correctly</b> among the total number of anamneses for which the disease was predicted. The higher the precision, the less frequently the algorithm will make errors in classifying elements.
+
+Recall shows the number of class elements correctly assigned to the class by the algorithm. In our case, it is the proportion of the anamneses correctly assigned to diabetes among the real number of diabetes cases in the sample. The higher the recall, the less frequently the algorithm misses the elements that need to be assigned to a certain class.
+</div>
+
+In the `utils.py` file, you will find the `train_test_split` function, which splits the given sample into training and control samples. The `plot_classification` function, which you will also find there, builds a graph showing the dependence of the classifier's error rate on the iteration. If you import this function in task.py, call it in main, and run the script, it will visualize the change of the error rate depending on the training rate and the loss function. The graphs for both functions will be saved as images. As we have already mentioned, the gradient descent with low values of training rate converges quite slowly, so be prepared to wait for the results.
+
+Currently, `plot_classification` builds three graphs for each loss function – with $1e-6$, $1e-5$, and $1e-4$ training rates respectively. As we've mentioned above, large values of $\alpha$ may result in "skipping the minimum" – at one step, the algorithm won't reach the stopping condition yet, but at the next step, it will start leaving the desired values behind. It will lead to a further decrease of the gradient and the margin and ultimately, result in the ["exploding gradients"](https://www.machinelearningmastery.ru/exploding-gradients-in-neural-networks/) problem. You can see this effect if you set $\alpha = 1e-3$. The margins and gradients will keep growing after passing the minimum until they cause an overflow in power in the loss function and all values will become indefinite. There are several solutions to the problem, but we won't discuss them in this course, as the problem will be solved when we modify the algorithm in the next task.
```

Comment:
skipping/jumping – both are my inventions here, couldn't find any creditable source. Let's stick to the one you find more natural
will have left – good
I'd say "cause an arithmetic overflow", otherwise good